tools/dds/dds-server/rs-dds-server.cpp

```diff
@@ -90,16 +96,122 @@ try
         return EXIT_FAILURE;
     }
 
+    std::map<rs2::device, std::pair<std::shared_ptr<librealsense::dds::dds_device_server>, std::shared_ptr<tools::lrs_device_manager>>> device_handlers_list;
     
     std::cout << "Start listening to RS devices.." << std::endl;
+
+    // Create a RealSense context
+    rs2::context ctx( "{"
+        "\"dds-discovery\" : false"
+        "}" );
+
     // Run the LRS device watcher
-    tools::lrs_device_watcher dev_watcher;
-    dev_watcher.run( [&]( rs2::device dev ) { broadcaster.add_device( dev ); },
-                     [&]( rs2::device dev ) { broadcaster.remove_device( dev ); } );
+    tools::lrs_device_watcher dev_watcher( ctx );
+    dev_watcher.run(
+        // Handle a device connection
+        [&]( rs2::device dev ) {
+
+            // Broadcast the new connected device to all listeners
+            auto dev_topic_root = broadcaster.add_device( dev );
+
+            // Create a supported streams list for initializing the relevant DDS topics
+            auto device_sensors = dev.query_sensors();
+            std::unordered_set<std::string> supported_streams_names;
+            for( auto sensor : device_sensors )
+            {
+                auto stream_profiles = sensor.get_stream_profiles();
+                std::for_each( stream_profiles.begin(),
+                               stream_profiles.end(),
+                               [&]( const rs2::stream_profile & sp ) {
+                                   if( supported_streams_names.find( sp.stream_name() )
```

Comment:
Why do we have to do this, if it's a `set`?

Reply:
Since the sensor has ~100 profiles I figured it will be more effective to search (average constant-time complexity) than to override ~100 times, but it's a minor optimization that I don't mind dropping :)

Reply:
Will be removed.

Reply:
Done